pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -509,11 +537,17 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(oldNodeStatus.status, node.Status) {
+			_, err = nc.kubeClient.Nodes().Update(node)
+			if err != nil {
+				glog.Errorf("Error updating node %s: %v", node.Name, err)
```

Comment:
Done. Sorry for the order of return types, I just had to add grace period and ready condition later, and it ended like this. For the record: I think that returning error as first value would be better, as it would strongly suggest that error should be checked before doing anything (it would be impossible to "forget" that there's a possibility of getting an error).